pkgs/by-name/nl/nlohmann_json_schema_validator/package.nix

```diff
@@ -0,0 +1,30 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, nlohmann_json
+, cmake
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "nlohmann_json_schema_validator";
+  version = "2.3.0";
+
+  outputs = [ "out" "dev" ];
+
+  src = fetchFromGitHub {
+    owner = "pboettch";
+    repo = "json-schema-validator";
+    rev = finalAttrs.version;
+    hash = "sha256-Ybr5dNmjBBPTYPvgorJ6t2+zvAjxYQISWXJmgUVHBVE=";
+  };
+
+  buildInputs = [ nlohmann_json ];
+  nativeBuildInputs = [ cmake ];
+
+  meta = {
+    description = "JSON schema validator for JSON for Modern C++";
+    homepage = "https://github.com/pboettch/json-schema-validator";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ br337 ];
+    platforms = lib.platforms.all;
+  };
```

Comment:
please add meta.mainProgram, since the pr adds multiple binarys and things like `lib.getExe` or `nix run` won't work without it.

Reply:
Unless there is a clear _single_ binary the package provides which is the one users expect to run, i oppose setting `mainProgram`. In such cases one is expected to use `lib.getExe'` or `nix shell PKG -c binary ...`.

Reply:
Being able to disable the `bin/${pname}` fallback behavior of `lib.getExe` with something like `mainProgram = null` would be great, but implementing that that is outside the scope of this PR

Reply:
> Unless there is a clear binary the package provides which is the one users expect to be run, i oppose setting mainProgram. In such cases one is expected to use lib.getExe' or nix shell PKG -c binary ....

I would have set it to `json-schema-validate` since that's the thing in the package name. But feel free to merge if you think otherwise, the package looks good to me except for this.

Reply:
So should I implement this change now or should I resolve the issue without further changes?